### PR TITLE
Reader: Add tests for classifyPost

### PR DESCRIPTION
**Pull request by blowery**
Incomplete, more needed for full coverage.

No functional changes (I hope). Just tests.

### CODE REVIEW

File: client/state/reader/posts/test/normalization-rules.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import { forEach, repeat } from 'lodash';
+import { expect } from 'chai';
+
+/**
+ * Internal Dependencies
+ */
+import { classifyPost } from '../normalization-rules';
+import * as DISPLAY_TYPES from '../display-types';
+
+function verifyClassification( post, displayTypes ) {
```
Comment (bluefuton) [inline client/state/reader/posts/test/normalization-rules.js]: Nice breakout 👍

File: client/state/reader/posts/test/normalization-rules.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	classifyPost( post );
+	forEach( displayTypes, displayType => {
+		expect( post.display_type & displayType ).to.equal( displayType );
+	} );
+}
+
+describe( 'normalization-rules', () => {
+	describe( 'classifyPost', () => {
+		it( 'empty post should get UNCLASSIFIED', () => {
```
Comment (bluefuton) [inline client/state/reader/posts/test/normalization-rules.js]: I'd reword to 'should mark an empty post UNCLASSIFIED'

File: client/state/reader/posts/test/normalization-rules.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+			verifyClassification( {}, [ DISPLAY_TYPES.UNCLASSIFIED ] );
+		} );
+
+		it( 'PHOTO_ONLY', () => {
```
Comment (bluefuton) [inline client/state/reader/posts/test/normalization-rules.js]: I'd reword to 'should classify a PHOTO_ONLY post'

File: client/state/reader/posts/test/normalization-rules.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			verifyClassification( {
+				canonical_media: {
+					mediaType: 'image',
+					width: 1000
+				},
+				better_excerpt_no_html: repeat( 'no ', 10 )
+			}, [ DISPLAY_TYPES.PHOTO_ONLY ] );
+		} );
+
+		it( 'no PHOTO_ONLY, content too long', () => {
```
Comment (bluefuton) [inline client/state/reader/posts/test/normalization-rules.js]: I'd reword to 'should not classify a PHOTO_ONLY post if the content is too long'

File: client/state/reader/posts/test/normalization-rules.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+			verifyClassification( {
+				canonical_media: {
+					mediaType: 'image',
+					width: 1000
+				},
+				better_excerpt_no_html: repeat( 'no ', 100 )
+			}, [ DISPLAY_TYPES.UNCLASSIFIED ] );
+		} );
+
+		it( 'CANONICAL_IN_CONTENT', () => {
```
Comment (bluefuton) [inline client/state/reader/posts/test/normalization-rules.js]: I'd reword to 'should classify a post as CANONICAL_IN_CONTENT'

File: client/state/reader/posts/test/normalization-rules.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+				},
+				content_images: [
+					{
+						src: 'http://example2.com/foo/bar/ping.jpg?w=20'
+					}
+				]
+			}, [ DISPLAY_TYPES.CANONICAL_IN_CONTENT ] );
+		} );
+
+		it( 'canonical image, but not CANONICAL_IN_CONTENT', () => {
```
Comment (bluefuton) [inline client/state/reader/posts/test/normalization-rules.js]: I'd reword to 'should not classify a post as CANONICAL_IN_CONTENT if the canonical image is not in the content'

File: client/state/reader/posts/test/normalization-rules.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+			verifyClassification( {
+				canonical_image: {
+					mediaType: 'image',
+					width: 1000,
+					uri: 'http://example.com/foo/baz/ping.jpg?w=2'
+				},
+				content_images: [
+					{
+						src: 'http://example2.com/foo/bar/ping.jpg?w=20'
```
Comment (bluefuton) [inline client/state/reader/posts/test/normalization-rules.js]: Maybe make this example URL more obviously different to the previous one, so it's easier to recognise the intent of the test?

**Comment by samouri** [comment]
sorry for the sidetracking, but I have a question on `classifyPost`.
Whats the tradeoff between having a function like `classifyPost` that creates the bitmask vs. a number of `isPostPhotoOnly( post )` style functions?

**Comment by bluefuton** [comment]
I've fixed my naming nits in 6b3e060.

@samouri do we need to make any alterations to the `CANONICAL_IN_CONTENT` tests given your recent changes in that department?

**Comment by samouri** [comment]
@bluefuton definitely. I'll make the changes

**Comment by samouri** [comment]
Had to do a force push because of a rebase

**Comment by bluefuton** [comment]
> Had to do a force push because of a rebase.
> Is there a preferred way?

Force push is all good in that scenario. If anyone else is actively working on the PR at the same moment I usually give them a heads up when I do it  👍

**Comment by bluefuton** [comment]
@samouri Ready for 👀 ?

**Comment by samouri** [comment]
Yep!
